src/plots/cartesian/tick_label_defaults.js

```diff
@@ -40,6 +40,7 @@ module.exports = function handleTickLabelDefaults(containerIn, containerOut, coe
 
         if(axType !== 'category') {
             var tickFormat = coerce('tickformat');
+            coerce('tickformatstops');
```

Comment:
Why not category?

Reply:
>Why not category?

Since `dtick` is always `1` for `category` axis type (at least all my tests show it), I still have no idea how `tickformatstops` should work for it 🤔  Maybe you have any thoughts?

Reply:
Right, `dtickrange` wouldn't make sense for category axes. I'm thinking for category axes each `tickformatstop` item would have:

```js
{
  ticktext: [/* */],
  tickvals: [/* */]
}
```

just like _regular_ axis-level `tickvals` and `ticktext` offering a solution for https://github.com/plotly/plotly.js/issues/1812

The _range_ of each `tickformatstop` is determined by the `tickvals[0]` and `tickvals[1]`.

My apologies, I thought I had written that down somewhere before :smirk:

Reply:
No problem, but not sure that I got everything right :) 
Let's confirm expected behaviour. For example:

I have the following categories:
`['a', 'b', 'c', 'd', 'e']`
https://codepen.io/anon/pen/brQwjG

then I define `tickformatstops`:

```
tickformatstops: [{
  ticktext: ['a-c'],
  tickvals: ['a', 'c']
}, {
  ticktext: ['d-e'],
  tickvals: ['d', 'e']
}]
```

and as result I get only 2 ticks:
`['a-c', 'd-e']`

Is it correct?

@talgalili

Reply:
Hi @apalchys 
This seems reasonable.
Question: does this need to be defined manually, or could it be automatically extracted from the values? For example, if we have 300 categories, that there would be a value setting to show only 30 categories at most. And if we are at a zoom level with many of the categories, it would spread them across the axis.
(I apologies if what I say makes little sense in this context. I more often deal with higher level implementations)

Reply:
@talgalili I see what you say but not sure how to fit it into the proposed API.  I am quite new in Plotly community.
@etpinard maybe you have any idea? 
Also is my assumption above correct? If yes, I can start the implementation.

Reply:
@apalchys checking about this further with @cpsievert - it appears that what I'm looking for is

> the ability to have something like nticks to be respected when tickmode='array'

(see the discussion here: https://github.com/talgalili/heatmaply/issues/78#issuecomment-310708211 )

Is there a way the current work would be extended to deal with this?

Reply:
I'll try to think some more about this issue at some point this week. Sorry for the wait.

Reply:
@etpinard have you had a chance to look into it?